services/flow-repository/app/api/controllers/mongo.js

```diff
@@ -119,6 +119,60 @@ const updateFlow = (storeFlow, credentials) => new Promise((resolve) => {
     });
 });
 
+const startFlow = (credentials, flowId) => new Promise((resolve) => {
+  const findId = mongoose.Types.ObjectId(flowId);
+
+  let qry;
+  if (credentials === 'admin') {
+    qry = { '_id': findId };
+  } else {
+    qry = {
+      $and: [{ '_id': findId },
+        { 'owners.id': { $in: credentials } },
+      ],
+    };
+  }
+
+  Flow.findOneAndUpdate(
+    qry,
+    { $set: { status: 'active' } },
+    { upsert: false, new: true },
+  )
+    .then((doc) => {
+      resolve(doc._doc);
+    })
+    .catch((err) => {
+      log.error(err);
+    });
+});
+
+const stopFlow = (credentials, flowId) => new Promise((resolve) => {
+  const findId = mongoose.Types.ObjectId(flowId);
+
+  let qry;
+  if (credentials === 'admin') {
+    qry = { '_id': findId };
+  } else {
+    qry = {
+      $and: [{ '_id': findId },
+        { 'owners.id': { $in: credentials } },
+      ],
+    };
+  }
+
+  Flow.findOneAndUpdate(
```

Comment:
All `mongoose` methods return promises. No need to use `new Promise`.

Reply:
Certain methods return thenables rather than full Promises, which has caused issues for us in the past. Hence why we've switched to manual wrapping.

Reply:
I guess it was quite a long time ago. Actually [this line](https://github.com/openintegrationhub/openintegrationhub/blob/master/services/integration-content-repository/app/server.js#L134) solves the problem. We are using only `await Model.method` and don't face any problems. But I'm not insisting on changing this file. Just want to share our experience with the subject.